File: app/src/main/java/com/smallcake/temp/http/BaseResponse.java
package com.smallcake.temp.http;

public class BaseResponse<T> {
    /** 返回码 */
    private int code;
    /** 提示语 */
    private String message;
    /** 数据 */
    private T data;

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public void setData(T data) {
        this.data = data;
    }

    public T getData() {
        return data;
    }
}

File: app/src/main/java/com/smallcake/temp/http/BaseResponse.kt
package com.smallcake.temp.http

/**
 * 后端返回数据基类
 * @param <T>
</T> */
class BaseResponse<T> {
    /** 返回信息  */
    var reason :String?= null
    var msg :String?= null
    /** 返回码  */
    var code :Int?= 0
    var resultcode :Int?= null
    var error_code :Int= 0

    /** 数据  */
    var result: T? = null
        private set


    fun setResult(result: T) {
        this.result = result
    }

}